README.md

```diff
@@ -1,85 +1,75 @@
 <!-- default badges list -->
-![](https://img.shields.io/endpoint?url=https://codecentral.devexpress.com/api/v1/VersionRange/454625952/21.1.6%2B)
 [![](https://img.shields.io/badge/Open_in_DevExpress_Support_Center-FF7200?style=flat-square&logo=DevExpress&logoColor=white)](https://supportcenter.devexpress.com/ticket/details/T1064526)
 [![](https://img.shields.io/badge/📖_How_to_use_DevExpress_Examples-e9f6fc?style=flat-square)](https://docs.devexpress.com/GeneralInformation/403183)
 <!-- default badges end -->
 
-# GridLookup for Web Forms - How to emulate a dropdown with grouped items
+# Grid Lookup for ASP.NET Web Forms - How to create a dropdown with grouped items
 <!-- run online -->
 **[[Run Online]](https://codecentral.devexpress.com/454625952/)**
 <!-- run online end -->
 
+The example demonstrates how to create a dropdown with grouped items. 
 
-The example illustrates how to create a dropdown with grouped items. 
+![image](drop-down-with-groups.png)
 
-![image](https://user-images.githubusercontent.com/69703500/152309609-d46559e1-f75a-4a28-aa90-3f3d4db712e6.png)
+## Implementation Details
 
+In this example, the client [ASPxClientGridLookup.KeyPress](https://docs.devexpress.com/AspNet/js-ASPxClientTextEdit.KeyPress) event is handled to implement incremental filtering. The event handler uses the nested grid's [ApplySearchPanelFilter](https://docs.devexpress.com/AspNet/js-ASPxClientGridView.ApplySearchPanelFilter(value)) method to filter data columns. 
 
-The implementation uses the nested grid's <a href="https://docs.devexpress.com/AspNet/js-ASPxClientGridView.ApplySearchPanelFilter(value)">ApplySearchPanelFilter</a> method to filter and search the data columns. The method is executed in the client-side **OnKeyPress** event of the GridLookup to simulate incremental filtering. Use the client-side <a href="https://docs.devexpress.com/AspNet/js-ASPxClientUtils.GetKeyCode.static(htmlEvent)">ASPxClientUtils.GetKeyCode</a> method to get the key pressed. To prevent multiple callback requests when keys are pressed in succession, use JavaScript's **setTimeout** and **clearTimeout** methods. Please note that the Enter key and Arrow keys will send a callback request when they are pressed in the OnKeyPress event. In this case, check for the key pressed in the event and prevent filtering to avoid sending callback requests.
+To prevent multiple callback requests when keys are pressed in succession, `setTimeout` and `clearTimeout` methods are used. 
 
-# JavaScript
+The Enter key and Arrow keys send a callback request when they are pressed in the `KeyPress` event. Call the [ASPxClientUtils.GetKeyCode](https://docs.devexpress.com/AspNet/js-ASPxClientUtils.GetKeyCode.static(htmlEvent)) method to get the key pressed and prevent filtering when these key are pressed to avoid sending callback requests.
```

Comment:
```suggestion
The Enter key and Arrow keys send a callback request when they are pressed in the `KeyPress` event. Call the [ASPxClientUtils.GetKeyCode](https://docs.devexpress.com/AspNet/js-ASPxClientUtils.GetKeyCode.static(htmlEvent)) method to get the key that was pressed and prevent filtering when these keys are pressed to avoid sending callback requests.
```